package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,262 @@
+and eigenvectors
+
+   >>> u = MDAnalysis.Universe(PSF,DCD)
+   >>> dmap = diffusionmap.DiffusionMap(u)
+   >>> dmap.run()
+   >>> eigenvalues = dmap.eigenvalues
+   >>> eigenvectors = dmap.eigenvectors
+
+Classes
+-------
+
+.. autoclass:: DiffusionMap
+   :members:
+
+References
+---------
+
+If you use this Dimension Reduction method in a publication, please
+reference:
+..[Lafon1] Coifman, Ronald R., Lafon, Stephane (2006) Diffusion maps.
+Appl. Comput. Harmon. Anal. 21, 5–30.
+..[Clementi1] Rohrdanz, M. A, Zheng, W, Maggioni, M, & Clementi, C. (2013)
+Determination of reaction coordinates via locally scaled
+diffusion map. Journal of Chemical Physics.
+
+.. If you choose the default metric, this module uses the fast QCP algorithm
+[Theobald2005]_ to calculate the root mean square distance (RMSD) between
+two coordinate sets (as implemented in :func:`MDAnalysis.lib.qcprot.CalcRMSDRotationalMatrix`).
+When using this module in published work please cite [Theobald2005]_.
+
+
+"""
+
+import logging
+import numpy as np
+import MDAnalysis.lib.qcprot as qcp
+from six.moves import range
+from .base import AnalysisBase
+
+logger = logging.getLogger("MDAnalysis.analysis.diffusionmap")
+
+
+class DiffusionMap(AnalysisBase):
+    """Non-linear dimension reduction method
+
+    Dimension reduction with diffusion map of the structures in the universe.
+
+    Attributes
+    ----------
+    atoms : AtomGroup
+        Selected atoms in trajectory subject to dimension reduction
+    diffusion_matrix : array
+        Array of all possible ij metric distances between frames in trajectory
+    eigenvalues: array
+        Eigenvalues of the diffusion map
+    eigenvectors: array
+        Eigenvectors of the diffusion map
+        The second and higher eigenvectors ev[i+1,:] represent the i-th slowest
+        collective coordinates.
+
+    """
+
+    def __init__(self, u,  select='all', epsilon='average', k=10, weights=None,
+                 metric=None, start=None, stop=None, step=None):
+        """
+        Parameters
+        -------------
+        u : trajectory `~MDAnalysis.core.AtomGroup.Universe`
+            The MD Trajectory for dimension reduction, remember that computational
+            cost scales at O(N^3) where N is the number of frames.
+            Cost can be reduced by increasing step interval or specifying a
+            start and stop
+        select: str, optional
+            1. any valid selection string for
+            :meth:`~MDAnalysis.core.AtomGroup.AtomGroup.select_atoms`
+            This selection of atoms is used to calculate the RMSD between
+            different frames. Water should be excluded.
+        epsilon : float, optional
+            Specifies the epsilon used for the diffusion map. More information
+            in [1] and [2]. With 'average' the average of the RMSD to the k-th
+            greatest value is used. (#TODO, check if actually agrees with theory)
```

Comment:
Yup this translates roughly to. 

> We played around with ε until we got some sensible results. Just use whatever seems appropriate to you and don't worry to much as the result hopefully won't depend much on ε.

For us this means we should allow to easily change the value of ε once a diffusion map is calculated. So we need to store the distance matrix where you apply some metric for each frame pair `ij` and allow explicit calling to `conclude` with an updated ε.